pxtlib/sprite-editor/gallery.ts

```diff
@@ -93,13 +93,22 @@ namespace pxtsprite {
                 .map(el => el.substring(1))
                 .filter(el => !!el);
 
-            return target.filter(el =>
-                includeTags.every(tag => el.tags.indexOf(tag) !== -1)
-                && excludeTags.every(tag => {
-                    const index = el.tags.indexOf(tag);
-                    return index === -1 || el.tags.charAt(index - 1) === "?";
-                })
-            );
+            return target.filter(el => checkInclude(el) && checkExclude(el));
+
+            function checkInclude(item: GalleryItem) {
+                return includeTags.every(filterTag =>
+                    item.tags.some(tag => {
+                        const ind = tag.indexOf(filterTag);
+                        return ind === 0 || ind === 1 && tag.charAt(0) === "?";
+                    })
+                )
+            }
+
+            function checkExclude(item: GalleryItem) {
+                return excludeTags.every(filterTag =>
+                    !item.tags.some(tag => tag.indexOf(filterTag) === -1)
```

Comment:
as a side note I phrased the commit message for this poorly, these are currently counting as matching the tags as long as there is a tag on the image that **starts with** the given tag; easy enough to change if preferred

Reply:
I think this should be an explicit equals check (ignoring case, but matching the whole string)